packages/scripts/src/flux-emulator/lib.ts

```diff
@@ -171,94 +150,80 @@ type IntegrationTestReducer = {
-        const { integrationTests } = reduced
-
-        try {
-          const parsed = JSON.parse(consoleOut)
-          if ('input' in parsed) latestInput = parsed.input
-          else if ('output' in parsed && latestInput) {
-            integrationTests.push(latestInput.data)
-            latestInput = null // Ensure we don't use the same input twice
-          }
-          return { latestInput, integrationTests }
-        } catch (e: any) {
-          return { latestInput, integrationTests }
+  logInfo('Running integration tests')
+
+  const integrationTestOutput = shell
+    .exec(`yarn test ${pathToAdapter}/test/integration/*.test.ts`, {
+      fatal: true,
+      silent: true,
+      env: { ...process.env, ...testEnvOverrides },
+    })
+    .toString()
+
+  const { integrationTestPayloads } = integrationTestOutput.split('\n').reduce(
+    (reduced: Record<string, any>, consoleOut) => {
+      let { latestInput } = reduced
+      const { integrationTestPayloads } = reduced
+
+      try {
+        const parsed = JSON.parse(consoleOut)
+        if ('input' in parsed) latestInput = parsed.input
+        else if ('output' in parsed && latestInput) {
+          integrationTestPayloads.push(latestInput)
+          latestInput = null // Ensures we don't use the same input twice
         }
-      },
-      { integrationTests: [] },
-    )
-
-    const integrationTestPayloads = integrationTests.map(
-      (data: Record<string, unknown>, i: number) => ({
-        name: `integration-${i}`,
-        data,
-      }),
-    )
-    configPayloads.push(...integrationTestPayloads)
-
-    // If no payloads from integration tests either, check test-payload.json
-    if (!configPayloads.length) {
-      logInfo('No payload found in integration tests, falling back to test-payload.json')
-
-      const payloadPath = pathToAdapter + '/test-payload.json'
-      if (shell.test('-f', payloadPath)) {
-        const testFile = JSON.parse(shell.cat(payloadPath).toString())
-        const testPayloads = testFile.requests.map((data: Record<string, unknown>, i: number) => ({
-          name: `test-payload-${i}`,
-          data,
-        }))
-        configPayloads.push(...testPayloads)
+        return { latestInput, integrationTestPayloads }
+      } catch (e) {
+        return { latestInput, integrationTestPayloads }
       }
-    }
+    },
+    { integrationTestPayloads: [] },
+  )
+
+  //TODO get names for the payloads (based on what the flux emulator names would be)
+  nameAndData.push(
+    ...integrationTestPayloads.map((data: Record<string, any>) => ({
+      name: 'integration-test',
+      data,
+    })),
+  )
+
+  const payloadPath = pathToAdapter + '/test-payload.json'
+  if (shell.test('-f', payloadPath)) {
+    const examplePayload = JSON.parse(shell.cat(payloadPath).toString())
+    //TODO get names for the payloads (based on what the flux emulator names would be)
+    nameAndData.push({ name: 'test-payload', data: examplePayload })
```

Comment:
Guessing these TODOs aren't critical?

Reply:
Right, I don't know how we intend to generate these payload names, and I also don't understand anything that could depend on it... I'll go ahead and remove the TODOs :sweat_smile: